### ISSUE TITLE
Add `glusterfs` storage driver

### ISSUE THREAD

**Issue by geaaru** [lxc/incus#344]
This is a complex feature but could be cool to have it considering that OpenStack in the past was working well with this driver and probably could be easier to configure of a Ceph Cluster.

Just my cent.

**Comment by adamcstephens** [comment]
I agree that glusterfs is a simpler solution and easier to configure than ceph. I would be a bit cautious about this effort at this time though. RedHat has discontinued their glusterfs storage offering and the last version is EOL in 12/2024. Development is already quite sparse and I would consider it a risk that it stops completely.

**Comment by geaaru** [comment]
:'( This is very bad news. A lot of opensource projects seem to have a hard life in the last period. Thank you for this update.

**Comment by stgraber** [comment]
Yeah, it's been on my list to look at for potential development, but we indeed won't be doing that if the project looks like it's about to be killed off...

There's also been sporadic interest in getting Linstor added as another option.

**Comment by adamcstephens** [comment]
Red Hat Gluster lifecycle link: https://access.redhat.com/support/policy/updates/rhs

**Comment by geaaru** [comment]
@adamcstephens btw, If Redhat does not support Glusterfs anymore doesn't mean that people from the community will not take care of it. yeah, maybe things will go slowly but I hope that someone will take care of it.
From the link seems that RedHat will not support anymore the Glusterfs product. The same happens for RedHat Tower (later renamed in Awx). :crossed_fingers:

**Comment by stgraber** [comment]
I haven't looked at the upstream repos yet, I think we'll just want to make sure that there are still going to be some maintainers around and that they intend to still make some new releases before spending too much time on it on our end.

We definitely don't need a crazy roadmap, if anything, a mature well working project is easier to deal with, we just want to make sure that the thing isn't going to go away and that if we somehow discover some bugs while working on this that there's going to be someone to fix them :)

**Comment by adamcstephens** [comment]
Yeah to be clear I'm just pointing out the risk. It's possible some maintainers will carry the torch forward, and stable isn't necessarily bad. :)

**Comment by htcosta** [comment]
I agree with @adamcstephens, i think glusterfs will most probably stale its development, and the effort to support glusterfs in incus could be then maybe prioritized to offer more active projects such as DRDB + LINSTORE.

**Comment by htcosta** [comment]
Glusterfs is not dead, but it's development has slow down due to different reasons, some of them already expressed on previous comments. Please consider closing this issue for now.

**Comment by bensmrs** [comment]
I kinda agree keeping this in the issue backlog isn’t really useful
